### PR TITLE
Upgrade to use jcasc test harness

### PR DESCRIPTION
**Pull request by casz**
JCasc had to create a new dependency that wasn't a tests classifier because maven doesn't allow you to bring dependencies through classifiers.

Upgrading here so that PCT works again,

Tracking:
jenkinsci/bom#164

Note: We'll need a release for PCT please

### CODE REVIEW

**Comment by casz** [comment]
all green 😄

File: pom.xml [hunk 1]
```diff
@@ -90,6 +90,12 @@
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <scope>test</scope>
+            <exclusions>
```
Comment (chriskilding) [inline pom.xml]: I discovered it's easier to just remove this junit:junit dependency - then you don't have to do the exclusion for hamcrest. Tested locally and it works. (Other dependencies have their own opinions on the junit and hamcrest versions, so it's easier to let them transitively pull in junit.)

**Comment by chriskilding** [comment]
One thing to change, otherwise looks ready to go

File: pom.xml (hunk 1 above)
Comment (casz) [reply pom.xml]: Thanks @chriskilding

**Pull request merged**